pkg/agent/agent_windows.go

```diff
@@ -218,6 +322,71 @@ func (i *Initializer) prepareOVSBridgeOnHNSNetwork() error {
 	return nil
 }
 
+func (i *Initializer) prepareOVSConfigForVM() error {
+	klog.InfoS("Performing OVS configuration", "hostIfName", i.nodeConfig.UplinkNetConfig.Name)
+	hostIfName := i.nodeConfig.UplinkNetConfig.Name
+	uplinkIfName := util.GenUplinkInterfaceName(hostIfName)
+	ovsPorts, ovsErr := i.ovsBridgeClient.GetPortList()
+	if ovsErr != nil {
+		klog.Errorf("Failed to list OVS ports")
+		return ovsErr
+	} else {
+		for index := range ovsPorts {
+			port := &ovsPorts[index]
+			if port.Name == hostIfName {
+				klog.InfoS("Uplink and host configuration exists in OVS")
+				return nil
+			}
+		}
+	}
+
+	success := false
+	uplinkExternalIDs := map[string]interface{}{
+		interfacestore.AntreaInterfaceTypeKey: interfacestore.AntreaUplink,
+	}
+	// Create uplink port on OVS.
+	uplinkUUID, ovsErr := i.ovsBridgeClient.CreatePort(uplinkIfName, uplinkIfName, uplinkExternalIDs)
+	if ovsErr != nil {
+		return fmt.Errorf("failed to create uplink port on OVS for %s: %v", uplinkIfName, ovsErr)
+	}
+
+	defer func() {
+		if !success {
+			klog.InfoS("Exiting!! delete port on OVS", "uplinkUUID", uplinkUUID)
+			i.ovsBridgeClient.DeletePort(uplinkUUID)
+		}
+	}()
+
+	// Query the uplink port to check if its created
+	uplinkOFPort, ovsErr := i.ovsBridgeClient.GetOFPort(uplinkIfName, false)
+	if ovsErr != nil {
+		return fmt.Errorf("failed to create uplink ofport on OVS for %s: %v", uplinkIfName, ovsErr)
+	}
+	klog.InfoS("Added uplink port on OVS", "port", uplinkOFPort)
+	attachInfo := externalnode.GetOVSAttachInfo(uplinkIfName, uplinkUUID, hostIfName, "", i.namespace, []string{""})
```

Comment:
Here you want to reset the ExternalEntity name in the interfaceConfig/OVSDB in external_node_controller? Add comment here to explain why entity name is "", and when will be set?